### DISCUSSION TITLE
Bug in json reporter line number?

### DISCUSSION THREAD

**Discussion by prismic** [vitest-dev/vitest#5350] [General]
Reproduction here: https://stackblitz.com/edit/vitest-dev-vitest-mfg2jr?file=package.json&view=editor

I am unsure if this intentional or a bug, so I thought I ask here first.

It seems there might be a bug in the line number reported by the json reporter, in that it is taking the first line of the stack trace and using the line number there, regardless whether that is the file under test.  This works fine in some cases, but when using something like testing-library and the error occurs in something like a `screen.getByText()`, the first line of the stack trace is coming from testing-library instead.

```
 FAIL  src/components/input.test.tsx > Input > should change input value
...
 ❯ Object.getElementError node_modules/@testing-library/dom/dist/config.js:37:19
 ❯ eval node_modules/@testing-library/dom/dist/query-helpers.js:76:38
 ❯ eval node_modules/@testing-library/dom/dist/query-helpers.js:52:17
 ❯ eval node_modules/@testing-library/dom/dist/query-helpers.js:95:19
 ❯ eval src/components/input.test.tsx:20:19
```
In the above, other reporters correctly highlight `input.test.tsx:20:19` as the line producing the failing test, but json reporter shows this:
 
 ```
 {
    "numTotalTestSuites": 2,
    "numPassedTestSuites": 2,
    ...
    "success": false,
    "testResults": [
        {
            "assertionResults": [
                {
                    "ancestorTitles": [
                        "",
                        "Input"
                    ],
                    "fullName": " Input should change input value",
                    "status": "failed",
                    "title": "should change input value",
                    "duration": 14,
                    "failureMessages": [
                        "..."
                    ],
                    "location": {
                        "line": 37,
                        "column": 19
                    }
                }
            ],
            ...
        }
    ]
}
```
 
Is this a legitimate bug?

**Comment by sheremet-va** [comment]
The location in the json reporter is just the first error location, it's not the same value that jest uses. It is a bug.
